### PR TITLE
fix js import test not passing

### PR DESCRIPTION
**Pull request by Plaristote**
* replace qmlEngine references with engine
* fix jsparser not properly walking the AST_Tree
* enable the QMLEngine JavascriptImport test

I noticed that js import wasn't working. Then I noticed the test was disabled. So I found what was blocking the test and fixed it (qmlEngine was used instead of engine in some place.... but qmlEngine is defined by the AutoLoader, and it's exclusively for use outside of qmlweb's scope. So I replaced the occurences of qmlEngine with engine).

Then I fixed a bug in jsparser, not properly using the AST_Tree returned by qmlparse.

The test was wrong as well: if we set a color to 'magenta', we shouldn't expect the CSS property to be 'magenta', but the rgb value for the magenta color (as this is the new behavior for QMLColor).

### CODE REVIEW

File: src/qtcore/qml/QMLEngine.js [hunk 1]
```diff
@@ -115,7 +115,6 @@ QMLEngine = function (element, options) {
 
         // Create and initialize objects
         var component = new QMLComponent({ object: tree, parent: null });
-
```
Comment (ChALkeR) [inline src/qtcore/qml/QMLEngine.js]: Unrelated style change, let's keep those minimal until we finish with the merges..

Comment (Plaristote) [inline src/qtcore/qml/QMLEngine.js]: Hadn't noticed that one. It's fixed.

**Comment by ChALkeR** [comment]
Those parser lines are from 5be63f5d.

Everything except for the parser LGTM, and I can't give any comments about the parser — I am not familiar with that code.

File: src/qtcore/qml/QMLEngine.js (hunk 1 above)
Comment (ChALkeR) [inline src/qtcore/qml/QMLEngine.js]: Thanks!

**Comment by ChALkeR** [comment]
On a second thought, this does LGTM. Merging.
